### DISCUSSION TITLE
Got invalid transaction key error when signing transactions

### DISCUSSION THREAD

**Discussion by jingxuan98** [WalletConnect/walletconnect-monorepo#2840] [Developer Support] [answered]
I am using https://react-wallet.walletconnect.com/ as a wallet to test with

But when i try to sign the pop up modal for the transaction in the react-wallet.walletconnect, i get this error which says that 

![Screenshot 2023-06-28 at 9 43 11 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/6fbd64a4-f3b7-44f9-a491-453d9ca6c6ba)

Any idea how to solve this? and any other wallets avai for testnet testing as well? i tried tokenPocket but not sure why it seems to just now able to scan walletconnect QR by web3modal

**Comment by finessevanes** [comment]
Hey, thanks for reporting. Could we get the following:
1) Version of the package you are using
2) Reproducible code via a codeBox, error logs, and/or reproduction steps

**Reply by jingxuan98** [reply, marked as answer]
guess its the wallet provider issue, switch to use metamask to test and it works fine
